relnotes/integrationtests.migration.md

```diff
@@ -0,0 +1,16 @@
+* Unit and integration tests changes
+
+  * Starting with v2.0.0 Makd will compile all modules from integration tests as part of regular unittest target and stop supplying `-unittest` flag when compiling actual integration tests. This may break compilation of a project so some manual tweaks are required:
+
+    1. If integration test modules don't have module statements or have one with identical module names, it will result in compilation failure.
+
+       The solution is to add explicit module statements to give each file distinct module name.
+
+    2. As with modules containing a `main()` in the regular user code in `$(SRC)`, integration test modules containing `main()` functions need to be modified to version out that `main()` function via `version(UnitTest)`, otherwise the unittest target will fail to link because of multiple `main()`s.
+
+       Again, you could manually add these modules to `TEST_FILTER_OUT`, but it is not recommended as any unit tests present in the module with the `main()` function won't be ran this way.
+
+  * Integration tests default location changed from `test` to `integrationtest`
+
+    The location can be set explicitly via the variable `INTEGRATIONTEST`, but you are not required to define it anymore (as it was required in v1.x.x), you just need to define it if you don't want the new default (`integrationtest`). This change was introduced to avoid symbol clashes in D2.
```

Comment:
"This change was introduced to avoid symbol clashes in D2." -- maybe explain this a little bit more? (Just from that sentence, I don't understand the reasoning.)

Reply:
Added the classical example, @mihails-strasuns-sociomantic please check if what I'm saying is correct, as I might be forgetting or misinterpreting some detail.

Reply:
I would rephrase it more to the point "can't directly use any symbol named `test` if you root package name is also `test`".